ignis/widgets/button.py

```diff
@@ -65,27 +66,41 @@ def on_right_click(self) -> Callable:
         """
         The function to call on right click.
         """
-        return self._on_right_click
+
+        def wrapper(*args, **kwargs):
+            if self._on_right_click is None:
+                return
+
+            return self._on_right_click(*args, **kwargs)
+
+        return wrapper
 
     @on_right_click.setter
     def on_right_click(self, value: Callable) -> None:
         self._on_right_click = value
         if not self.__right_click_controller:
             self.__right_click_controller = self.__init_controller(
-                3, self._on_right_click
+                3, self.on_right_click
```

Comment:
Hmm, what about using lambda functions instead of creating wrappers in getter functions?
```suggestion
                3, lambda *args, **kwargs: self._on_right_click(*args, **kwargs)
```
I think It looks a bit cleaner

Reply:
you are right, that does look cleaner, I overengineered a bit here

Reply:
I pushed a version with lambdas but used attributes without leading `_`, just to rely on the "proper" external values (in case getter will change some day for whatever reason?)